PlayCover/Views/Settings/IPASourceSettings.swift

```diff
@@ -132,12 +140,23 @@ struct SourceView: View {
             case .empty:
                 EmptyView()
             case .valid:
-                StatusBadgeView(imageName: "checkmark.circle.fill",
-                                imageColor: .green,
-                                popoverText: "preferences.popover.valid",
-                                showingPopover: $showingPopover)
+                if isEnabled {
+                    StatusBadgeView(imageName: "checkmark.circle.fill",
+                                    imageColor: .green,
+                                    popoverText: "preferences.popover.valid",
+                                    showingPopover: $showingPopover)
+                } else {
+                    StatusBadgeView(imageName: "checkmark.circle.badge.xmark.fill",
+                                    imageColor: .gray,
+                                    popoverText: "state.disabled",
+                                    showingPopover: $showingPopover)
+                }
             }
         }
+        .onChange(of: isEnabled) { value in
+            StoreVM.shared.enableSourceToggle(source: source, value: value)
+            StoreVM.shared.updateSourcesApps()
```

Comment:
When the user toggles the source while on the IPA Library tab, it causes PlayCover to stutter as it attempts to load and render all the apps. This should be fixed by simply putting this function inside a `Task`. You might need to store the task however to prevent weird behaviour when the user spams the toggle (maybe implement something similar to `resolveTask` in StoreVM. Or alternatively, you can disable the UI while the function is updating the available sources.